src/ray/gcs/gcs_server/gcs_actor_manager.h

```diff
@@ -45,6 +45,14 @@ class GcsActor {
   explicit GcsActor(rpc::ActorTableData actor_table_data)
       : actor_table_data_(std::move(actor_table_data)) {}
 
+  /// Create a GcsActor by actor_table_data and task_spec.
+  ///
+  /// \param actor_table_data Data of the actor (see gcs.proto).
+  /// \param task_spec Task spec of the actor.
+  explicit GcsActor(rpc::ActorTableData actor_table_data, rpc::TaskSpec task_spec)
+      : actor_table_data_(std::move(actor_table_data)),
+        task_spec_(std::make_unique<rpc::TaskSpec>(task_spec)) {}
```

Comment:
Can you add RAY_CHECK(state != DEAD) in the body?

Reply:
done